errors/errors.go

```diff
@@ -78,23 +77,12 @@ func (e Error) New(description string) error {
 	return Wrap(e, description)
 }
 
-// This Wrap implementation provides a transition layer between two Wrap
-// function implementations with incompatible notations.
-//
-// Once migration is complete, it will be removed and replaced by wrapng
-// function.
-func Wrap(err error, description ...string) TMError {
-	if len(description) == 0 {
-		return deprecatedLegacyWrap(err)
-	}
-	return wrapng(err, strings.Join(description, ", "))
-}
 
 // Wrap extends given error with an additional information.
 //
 // If the wrapped error does not provide ABCICode method (ie. stdlib errors),
 // it will be labeled as internal error.
-func wrapng(err error, description string) TMError {
+func Wrap(err error, description string) TMError {
```

Comment:
Note that the "legacy" Wrap also handles nil error... which is quite useful to avoid if statements everywhere. If you add

```go
if err == nil {
  return nil
}
```

to the top of this function, this makes it more flexible. Note the conversation here: https://github.com/iov-one/weave/pull/305#discussion_r255043779

Something like

```golang
if err := bnsClient.BroadcastTx(&tx).IsError(); err != nil {
  return fmt.Errorf("cannot broadcast transaction: %s", err)
}
return nil
```

in old code, could be simplified to

```golang
return errors.Wrap(
  bnsClient.BroadcastTx(&tx).IsError(), 
  "cannot broadcast transaction"))
```

Whether you want to use such a shortcut is a personal decision based on where it is being returned, but I think it is quite helpful for the library to allow such usage